src/index.ts

```diff
@@ -368,6 +368,27 @@ export class YouTube {
 
     return items
   }
+
+  private getId (input: string, type: string) {
+    if (input.includes('youtube.com') || input.includes('youtu.be')) {
+      const result = parseUrl(input)[`${type}`]
+      if (!result.startsWith('UC') && type === 'channel') {
+        return request.api('search', { q: result, type: 'channel', part: 'id' }, this.token, this.tokenType).then((r) => r.items[0].id.channelId)
```

Comment:
`q` could be `input`, not `result`.

Reply:
should*

Reply:
Also remove the parentheses around `r`, and the same for all other single-parameter lambda functions

Reply:
`items[0]` may be undefined, so if it is falsey return undefined. Do this for all of these.

Reply:
Actually, I'm wrong. `q` should be `result`.